### PR TITLE
[Merged by Bors] - chore: move utilities into Util directory

### PR DESCRIPTION
**Pull request by kim-em**
As discussed during review of #5513, and on [zulip](https://leanprover.zulipchat.com/#narrow/stream/270676-lean4/topic/lean_lib.20in.20subdirectory.3F/near/373604052)

- [x] depends on: #5513

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#5513
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Util/Cache/Requests.lean [hunk 1]
```diff
@@ -28,8 +28,8 @@ section Get
 
 /-- Formats the config file for `curl`, containing the list of files to be downloaded -/
 def mkGetConfigContent (hashMap : IO.HashMap) : IO String := do
-  -- We sort the list so that the large files in `MathlibExtras` are requested first.
-  hashMap.toArray.qsort (fun ⟨p₁, _⟩ ⟨_, _⟩ => p₁.components.head? = "MathlibExtras")
+  -- We sort the list so that the large files in `Util/TacticCaches` are requested first.
+  hashMap.toArray.qsort (fun ⟨p₁, _⟩ ⟨_, _⟩ => p₁.components.take 2 = ["Util ", "TacticCaches"])
```
Comment (digama0) [inline Util/Cache/Requests.lean:32]:
```suggestion
  hashMap.toArray.qsort (fun ⟨p₁, _⟩ ⟨_, _⟩ => p₁.components.take 2 = ["Util", "TacticCaches"])
```

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#5513~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**